src/export/StructureDefinitionExporter.ts

```diff
@@ -484,6 +484,21 @@ export class StructureDefinitionExporter implements Fishable {
     cleanResource(structDef, (prop: string) => prop == 'elements' || prop.indexOf('_') == 0);
     structDef.inProgress = false;
 
+    // check for another structure definition with the same id
+    // see https://www.hl7.org/fhir/resource.html#id
+    // the structure definition has already been added to the package, so it's fine if it matches itself
+    if (
+      this.pkg.profiles.some(profile => structDef.id === profile.id && structDef !== profile) ||
+      this.pkg.extensions.some(
+        extension => structDef.id === extension.id && structDef !== extension
+      )
```

Comment:
I don't think you can just search the `pkg` in this case. Because it is possible that I am using the bring your own JSON functionality to add an existing StructureDefinition with id `fish-patient`. Then I could declare a Profile in my FSH with id `fish-patient`. Based on my testing, that does not get caught, and the FSH exported profile is silently overwritten by the BYO-JSON profile. I think this could also be a problem in the ValueSetExporter, CodeSystemExporter, and InstanceExporter as well.

Reply:
Allowing BYOJ to overwrite FSH-generated definitions is currently an intentional feature.  This gives authors a way to "patch" files that SUSHI might generate incorrectly.  The idea is that once SUSHI fixes whatever bug, the BYOJ patch can be removed and it's all good.  I'm not sure this is the best feature ever, but it's the way things are meant to work for now anyway.  So it's fine not to issue an error on that.